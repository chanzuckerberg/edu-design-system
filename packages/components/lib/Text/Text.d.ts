import React from "react";
import { TypographyProps } from "../common/typography";
declare type TextElement = "p" | "span";
declare type Props = {
    /**
     * Controls whether to render text inline (defaults to "p");
     */
    as?: TextElement;
    children: TypographyProps<TextElement>["children"];
    className?: TypographyProps<TextElement>["className"];
    color?: TypographyProps<TextElement>["color"];
    size?: TypographyProps<TextElement>["size"];
    spacing?: TypographyProps<TextElement>["spacing"];
    tabIndex?: number;
    weight?: TypographyProps<TextElement>["weight"];
};
declare const Text: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLElement>>;
export default Text;
