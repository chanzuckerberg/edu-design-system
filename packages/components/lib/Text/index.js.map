{"version":3,"file":"index.js","sourceRoot":"","sources":["../../src/Text/index.ts"],"names":[],"mappings":";;;;;;AAAA,+BAAiC;AAAxB,gHAAA,OAAO,OAAA"}