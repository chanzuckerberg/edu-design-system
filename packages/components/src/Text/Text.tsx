import React, { ReactNode } from "react";
import {
  TypographyColor,
  TypographySize,
  styleFromColor,
  styleFromSize,
} from "../util/typography";
import tw, { styled } from "twin.macro";

interface BaseTextProps {
  bold: boolean;
  color?: TypographyColor;
  size: TypographySize;
}

const TextComponent = styled.p<BaseTextProps>(({ bold, color, size }) => [
  bold ? tw`font-bold` : tw`font-normal`,
  styleFromColor(color),
  styleFromSize(size),
]);

type Props = {
  /**
   * Specifies font weight as either bold or normal (defaults to normal).
   * TODO: We may add stricter enforcement of mutually exclusive props.
   */
  bold?: boolean;
  /**
   * The text content to present.
   */
  children: ReactNode;
  /**
   * The color of the text element. If no color provided, defaults to a base color.
   */
  color?: TypographyColor;
  /**
   * The size of the html element. If used as a Heading and no `as` prop is provided,
   * then the component uses this value to determine which html tag to render
   * (e.g. 'h1', 'h2', etc.)
   */
  size: TypographySize;
};

function Text(props: Props) {
  const {
    bold = false,
    children,
    size,
    /**
     * Components that wrap typography sometimes requires props such as event handlers
     * to be passed down into the element. One example is the tooltip component.  It
     * attaches a onHover and onFocus event to the element to determine when to
     * trigger the overlay.
     */
    ...rest
  } = props;

  return (
    <TextComponent bold={bold} size={size} {...rest}>
      {children}
    </TextComponent>
  );
}

export default Text;
