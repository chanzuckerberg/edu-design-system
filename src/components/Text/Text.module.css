@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    #TEXT
\*------------------------------------*/

/**
 * Remove top and bottom margins applied by browser for text.
 */
.text {
  margin-bottom: 0;
  margin-top: 0;
}

/**
 * Sizes
 */
.text--body,
.text--md {
  @mixin eds-theme-typography-body-text-md;
}
.text--lg {
  @mixin eds-theme-typography-body-text-lg;
}
.text--sm {
  @mixin eds-theme-typography-body-text-sm;
}
.text--xs {
  @mixin eds-theme-typography-body-text-xs;
}
.text--caption {
  @mixin eds-theme-typography-caption-text-md;
}
.text--overline {
  @mixin eds-theme-typography-kicker;
}

/**
 * Variant
 */
.text--error {
  color: var(--eds-theme-color-text-utility-error);
}
.text--base {
  color: var(--eds-theme-color-text-neutral-default);
}
.text--brand {
  color: var(--eds-theme-color-text-brand-primary);
}
.text--info {
  /* TODO(brand-tokens): remove */
  color: var(--eds-color-info-700);
}
.text--inherit {
  color: inherit;
}
.text--neutral {
  /* TODO(brand-tokens): update to text-neutral-default when new colors land */
  color: var(--eds-theme-color-text-neutral-subtle);
}
.text--success {
  color: var(--eds-theme-color-text-utility-success);
}
.text--warning {
  color: var(--eds-theme-color-text-utility-warning);
}
.text--white {
  color: var(--eds-theme-color-text-neutral-default-inverse);
}

/**
 * Weights
 */
.text--normal-weight {
  font-weight: var(--eds-font-weight-light);
}
.text--bold-weight {
  font-weight: var(--eds-font-weight-bold);
}

/**
 * Spacing
 */
.text--half-spacing {
  margin-bottom: var(--eds-size-half);
}

.text--1x-spacing {
  margin-bottom: var(--eds-size-1);
}

.text--2x-spacing {
  margin-bottom: var(--eds-size-2);
}

/**
 * 1) A passage of text, including various components (i.e. article, blog post)
 */
.text-passage {
  /**
	 * Paragraph within text passage
	 */
  p {
    margin-bottom: var(--eds-size-3);
  }

  /**
   * First child of a text passage
   * 1) Zero out top margin to avoid unintended
   *    gaps when leading with a heading or other
   *    element that has a top margin
   */
  > :first-child {
    margin-top: 0; /* 1 */
  }

  /**
   * Last child of a text passage
   * 1) Zero out bottom margin to allow the container
   *    element to control the bottom margin
   */
  > :last-child {
    margin-bottom: 0;
  }

  /**
	 * Link within the text passage
	 */
  a {
    @mixin textLink;
  }

  /**
	  * First-level heading within text passage
	  */
  h1 {
    @mixin eds-theme-typography-headline-lg;
  }

  /**
	 * Second-level heading within text passage
	 */
  h2 {
    @mixin eds-theme-typography-headline-md;
    margin-top: var(--eds-size-5);
  }

  /**
	 * Third-level heading within text passage
	 */
  h3 {
    @mixin eds-theme-typography-headline-sm;
    margin-top: var(--eds-size-5);
  }

  /**
	 * Fourth-level heading within text passage
	 */
  h4 {
    @mixin eds-theme-typography-title-md;
    margin-top: var(--eds-size-5);
  }

  /**
	 * Fifth-level heading within text passage
	 */
  h5 {
    @mixin eds-theme-typography-title-sm;
    margin-top: var(--eds-size-5);
  }

  /**
	 * Sixth-level heading within text passage
	 */
  h6 {
    @mixin eds-theme-typography-body-text-sm-bold;
    margin-top: var(--eds-size-5);
  }

  /**
	 * Unordered list within text passage
	 */
  ul {
    @mixin reset;
    list-style: disc;
    margin-left: var(--eds-size-2);
    margin-bottom: var(--eds-size-3);
  }

  /**
	 * Ordered list within text passage
	 */
  ol {
    @mixin reset;
    list-style: decimal;
    margin-left: var(--eds-size-2);
    margin-bottom: var(--eds-size-3);
  }

  /**
	 * List item within text passage
	 */
  li {
    margin-bottom: var(--eds-size-1);
  }

  /**
   * Blockquote within text passage
   */
  blockquote {
    font-style: italic;
    border-left-width: var(--eds-theme-border-width);
    border-left-style: solid;
    border-left-color: var(--eds-theme-color-border-brand-primary-strong);
    padding-left: var(--eds-size-2);
    margin-left: 0;
    margin-bottom: var(--eds-size-3);
  }

  /**
   * Horizontal rule
   */
  hr {
    border-width: var(--eds-theme-border-width);
    border-style: solid;
    border-color: var(--eds-theme-color-border-neutral-subtle);
    margin: var(--eds-size-3) 0;
  }
}

/**
 * Small text passage
 */
.text-passage--sm {
  li {
    margin-bottom: 0;
  }
}

.text-passage--lg {
  ul {
    margin-left: var(--eds-size-3);
  }
  ol {
    margin-left: var(--eds-size-3);
  }
}
