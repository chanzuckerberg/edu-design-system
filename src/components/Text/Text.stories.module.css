/* ------------------------------------*\
    #TEXT STORIES
\*------------------------------------ */

/**
 * 1) For the "variant" story to display the different text variant options.
 * 2) Uses grid to neatly align the variant variations.
 */
.variant__table {
  /* 1 */
  display: grid; /* 2 */
  grid-template-columns: 1fr 1fr 1fr;

  /* TODO: uncomment and replace styling once tokens have moved over */

  /* gap: var(--eds-size-half); */
  gap: 0.25rem;
}

.variant__item {
  height: 4rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

/**
 * Variants are segmented into left, middle, and right columns.
 * Background variants were chosen to represent typical body backgrounds.
 */
.variant__item-white {
  grid-column: 1; /* 1 */

  /* TODO: uncomment and replace styling once tokens have moved over */

  /* background-color: var(--eds-theme-color-background-neutral-default); */
  @apply bg-white;
}

.variant__item-light {
  grid-column: 2; /* 1 */

  /* TODO: uncomment and replace styling once tokens have moved over */

  /* background-color: var(--eds-theme-color-body-background); */
  @apply bg-neutral-200;
}

.variant__item-dark {
  grid-column: 3; /* 1 */

  /* TODO: uncomment and replace styling once tokens have moved over */

  /* background-color: var(--eds-theme-color-body-background-inverted); */
  @apply bg-neutral-700;
}
