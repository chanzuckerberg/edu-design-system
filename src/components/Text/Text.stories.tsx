import type { StoryObj, Meta } from '@storybook/react';
import React from 'react';

import { Text } from './Text';

export default {
  title: 'Components/Text',
  component: Text,
  parameters: {
    layout: 'centered',
    badges: ['api-2.0', 'theme-2.0'],
  },
  argTypes: {
    children: {
      control: {
        type: 'text',
      },
    },
  },
  decorators: [(Story) => <div className="m-spacing-size-2">{Story()}</div>],
} as Meta<typeof Text>;

type Story = StoryObj<typeof Text>;

export const Default: Story = {
  args: {
    children: 'Default <Text /> rendering',
  },
};

export const BodyXLarge: Story = {
  args: {
    preset: 'body-xl',
    children: 'Body extra-large',
  },
};

export const BodyLarge: Story = {
  args: {
    preset: 'body-lg',
    children: 'Body large',
  },
};

export const BodyMedium: Story = {
  args: {
    preset: 'body-md',
    children: 'Body medium',
  },
};
export const BodySmall: Story = {
  args: {
    preset: 'body-sm',
    children: 'Body small',
  },
};

export const BodyXSmall: Story = {
  args: {
    preset: 'body-xs',
    children: 'Body Xsmall',
  },
};

export const CaptionMedium: Story = {
  args: {
    preset: 'caption-md',
    children: 'Caption medium',
  },
};

export const OverridingFontFamily: Story = {
  args: {
    preset: 'body-md',
    children:
      'You can use utility classes to override the font family used for a given size',
    className: '!font-secondary',
  },
};

/**
 * Here we demonstrate how to use utility classes to augment the text.
 * Note that when present, `preset` will override the deprecated props.
 */
export const UsingColorTokens: Story = {
  render: (args) => (
    <div>
      <Text {...args} className="text-utility-warning" preset="body-xl">
        using <code>text-utility-warning</code> utility class
      </Text>
      <Text {...args} className="text-utility-favorable" preset="body-lg">
        using <code>text-utility-favorable</code> utility class
      </Text>
      <Text {...args} className="text-utility-critical" preset="body-md">
        using <code>text-utility-critical</code> utility class
      </Text>
    </div>
  ),
};
