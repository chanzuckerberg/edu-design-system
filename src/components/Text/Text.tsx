import React from 'react';

export type Size =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'body'
  | 'sm'
  | 'xs'
  | 'caption'
  | 'overline';

export type Color =
  | 'alert'
  | 'base'
  | 'brand'
  | 'yellow'
  | 'info'
  | 'inherit'
  | 'neutral'
  | 'success'
  | 'warning'
  | 'white';

export type Props = {
  /**
   * Controls whether to render text inline (defaults to "p");
   */
  as?: 'p' | 'span';
  text: React.ReactNode;
  className?: string;
  color?: Color;
  size?: Size;
  spacing?: 'none' | 'half' | '1x' | '2x';
  tabIndex?: number;
  weight?: 'bold' | 'normal' | null;
} & React.HTMLAttributes<HTMLElement>;

/**
 * ```ts
 * import {Text} from "@chanzuckerberg/eds";
 * ```
 */
export const Text = ({
  as = 'p',
  text,
  size = 'body',
  /**
   * Components that wrap typography sometimes requires props such as event handlers
   * to be passed down into the element. One example is the tooltip component.  It
   * attaches a onHover and onFocus event to the element to determine when to
   * trigger the overlay.
   */ ...rest
}: Props) => {
  const TagName = as;
  return (
    <TagName size={size} {...rest}>
      {text}
    </TagName>
  );
};

export default Text;
