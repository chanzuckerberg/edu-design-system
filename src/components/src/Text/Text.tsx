import React, { forwardRef } from "react";

import Typography, { TypographyProps } from "../common/typography";

type TextElement = "p" | "span";

type Props = {
  /**
   * Controls whether to render text inline (defaults to "p");
   */
  as?: TextElement;
  children: TypographyProps<TextElement>["children"];
  className?: TypographyProps<TextElement>["className"];
  color?: TypographyProps<TextElement>["color"];
  size?: TypographyProps<TextElement>["size"];
  spacing?: TypographyProps<TextElement>["spacing"];
  tabIndex?: number;
  weight?: TypographyProps<TextElement>["weight"];
} & React.HTMLAttributes<HTMLElement>;

/**
 * ```ts
 * import {Text} from "@chanzuckerberg/eds-components";
 * ```
 */
const Text = forwardRef<HTMLElement, Props>(
  (
    {
      as,
      children,
      size = "body",
      /**
       * Components that wrap typography sometimes requires props such as event handlers
       * to be passed down into the element. One example is the tooltip component.  It
       * attaches a onHover and onFocus event to the element to determine when to
       * trigger the overlay.
       */ ...rest
    }: Props,
    ref,
  ) => (
    <Typography as={as || "p"} ref={ref} size={size} {...rest}>
      {children}
    </Typography>
  ),
);

Text.displayName = "Text"; // Satisfy eslint.

export default Text;
