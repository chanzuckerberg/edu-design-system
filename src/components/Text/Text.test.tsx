import { generateSnapshots } from '@chanzuckerberg/story-utils';
import { render, screen } from '@testing-library/react';

import userEvent from '@testing-library/user-event';
import React from 'react';
import { Text } from './Text';
import * as TextStoryFile from './Text.stories';

describe('<Text />', () => {
  generateSnapshots(TextStoryFile);

  it('should add the passthrough className', () => {
    render(
      <Text className="passthrough" size="body">
        Some Text
      </Text>,
    );
    expect(screen.getByText('Some Text').classList).toContain('passthrough');
  });

  it('should handle refs', async () => {
    const HelperComponent = ({ as }: { as: 'p' | 'span' }) => {
      const refContainer = React.useRef(null);
      const onButtonClick = () => {
        expect(refContainer.current).toBe(
          screen.getByText(`Ref container parent test ${as}`),
        );
      };
      return (
        <>
          <Text as={as} ref={refContainer}>
            Ref container parent test {as}
          </Text>
          <button onClick={onButtonClick}>Test ref</button>
        </>
      );
    };
    const { rerender } = render(<HelperComponent as="p" />);
    expect(screen.getByText('Ref container parent test p')).toBeTruthy();
    userEvent.click(screen.getByRole('button'));

    rerender(<HelperComponent as="span" />);
    expect(screen.getByText('Ref container parent test span')).toBeTruthy();
    userEvent.click(screen.getByRole('button'));
  });

  it('should console warn if attempting to use "info" variant', () => {
    const consoleWarnMock = jest
      .spyOn(global.console, 'warn')
      .mockImplementation();
    render(<Text variant="info">Some Text 2</Text>);
    expect(consoleWarnMock).toHaveBeenCalledTimes(1);
    expect(consoleWarnMock).toHaveBeenCalledWith(
      'Info variant is deprecated, please consider another variant.',
    );
    consoleWarnMock.mockRestore();
  });
});
