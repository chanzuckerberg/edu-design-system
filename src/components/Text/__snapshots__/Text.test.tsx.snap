// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Body story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--body"
  >
    Body paragraph
  </p>
</div>
`;

exports[`<Text /> BodyLarge story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--lg"
  >
    Body large
  </p>
</div>
`;

exports[`<Text /> BodyMedium story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--md"
  >
    Body medium
  </p>
</div>
`;

exports[`<Text /> BodySmall story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--sm"
  >
    Body small
  </p>
</div>
`;

exports[`<Text /> BodyVariantSuccessBold story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--body text--success text--bold-weight"
  >
    Success variant body text, bold
  </p>
</div>
`;

exports[`<Text /> BodyXSmall story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--xs"
  >
    Body Xsmall
  </p>
</div>
`;

exports[`<Text /> Callout story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--callout"
  >
    Callout
  </p>
</div>
`;

exports[`<Text /> Caption story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--caption"
  >
    Caption
  </p>
</div>
`;

exports[`<Text /> Overline story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--overline"
  >
    Overline
  </p>
</div>
`;

exports[`<Text /> Spacing story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div>
    <p
      class="spacing text text--body text--half-spacing"
    >
      Spacing Half
    </p>
    <p
      class="spacing text text--body text--1x-spacing"
    >
      Spacing 1x
    </p>
    <p
      class="spacing text text--body text--2x-spacing"
    >
      Spacing 2x
    </p>
    <p
      class="spacing text text--body"
    >
      Bottom text to show spacing for spacing 2x
    </p>
  </div>
</div>
`;

exports[`<Text /> TextPassage story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="layout-linelength-container text text--body text-passage text-passage--body"
  >
    <h1>
      Heading 1
    </h1>
    <p>
      A text passage contains arbitrary text that might come from a CMS. It should live within a container that caps the line length of the text to avoid a straining reading experience.
    </p>
    <h2>
      Heading 2
    </h2>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <ul>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ul>
    <h3>
      Heading 3
    </h3>
    <ol>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ol>
    <p>
      <a
        href="/"
      >
        Lorem ipsum dolor sit amet
      </a>
      , consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <blockquote>
      <p>
        This is a quotation from something.
      </p>
      <cite>
        Cite source
      </cite>
    </blockquote>
    <h4>
      Heading 4
    </h4>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <hr />
    <p>
      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <h5>
      Heading 5
    </h5>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <p>
      That is all.
    </p>
  </div>
</div>
`;

exports[`<Text /> TextPassageLarge story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="layout-linelength-container text text--lg text-passage text-passage--lg"
  >
    <h1>
      Heading 1
    </h1>
    <p>
      A text passage contains arbitrary text that might come from a CMS. It should live within a container that caps the line length of the text to avoid a straining reading experience.
    </p>
    <h2>
      Heading 2
    </h2>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <ul>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ul>
    <h3>
      Heading 3
    </h3>
    <ol>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ol>
    <p>
      <a
        href="/"
      >
        Lorem ipsum dolor sit amet
      </a>
      , consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <blockquote>
      <p>
        This is a quotation from something.
      </p>
      <cite>
        Cite source
      </cite>
    </blockquote>
    <h4>
      Heading 4
    </h4>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <hr />
    <p>
      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <h5>
      Heading 5
    </h5>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <p>
      That is all.
    </p>
  </div>
</div>
`;

exports[`<Text /> TextPassageSmall story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="layout-linelength-container text text--sm text-passage text-passage--sm"
  >
    <h1>
      Heading 1
    </h1>
    <p>
      A text passage contains arbitrary text that might come from a CMS. It should live within a container that caps the line length of the text to avoid a straining reading experience.
    </p>
    <h2>
      Heading 2
    </h2>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <ul>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ul>
    <h3>
      Heading 3
    </h3>
    <ol>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
      <li>
        Here is a unordered list item
      </li>
    </ol>
    <p>
      <a
        href="/"
      >
        Lorem ipsum dolor sit amet
      </a>
      , consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <blockquote>
      <p>
        This is a quotation from something.
      </p>
      <cite>
        Cite source
      </cite>
    </blockquote>
    <h4>
      Heading 4
    </h4>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <hr />
    <p>
      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <h5>
      Heading 5
    </h5>
    <p>
      This is another paragraph of text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <p>
      That is all.
    </p>
  </div>
</div>
`;

exports[`<Text /> TextVariantInherit story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <p
    class="text text--body text--error"
  >
    This text surrounds the 
    <span
      class="text text--body text--inherit"
    >
      Child Text
    </span>
     and shows it should inherit variant from the parent
  </p>
</div>
`;
